Compilation fails on runner `macos-14`

Some relevant lines

> Downloading https://ghcr.io/v2/homebrew/core/gmp/manifests/6.3.0
> Already downloaded: /Users/runner/Library/Caches/Homebrew/downloads/70a72a71216843d66a953c06ff6337445ce9bc94fae9f0e301e2f59005274a8e--gmp-6.3.0.bottle_manifest.json
> ==> Fetching gmp
> ==> Downloading https://ghcr.io/v2/homebrew/core/gmp/blobs/sha256:78e4f40cba6419cf7e2d81e9c945d1e93744511bd5230bdfac1b69ed894914b4
> Already downloaded: /Users/runner/Library/Caches/Homebrew/downloads/96a5c6ba7b30c6f2734be44067ba66ef0de5e0973c913659c7b555a4b063029c--gmp--6.3.0.arm64_sonoma.bottle.tar.gz
> ==> Reinstalling gmp 
> ==> Pouring gmp--6.3.0.arm64_sonoma.bottle.tar.gz
> 🍺  /opt/homebrew/Cellar/gmp/6.3.0: 21 files, 3.3MB
> ==> Upgrading 1 dependent of upgraded formula:
> Disable this behaviour by setting HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK.

[…]

> Warning:  libtoolize does not appear to be available.  This means that
> the automatic build preparation via autoreconf will probably not work.
> Preparing the build by running each step individually, however, should
> work and will be done automatically for you if autoreconf fails.

> Fortunately, glibtoolize was found which means that your system may simply
> have a non-standard or incomplete GNU Autotools install.  If you have
> sufficient system access, it may be possible to quell this warning by
> running:

>    sudo ln -s /opt/homebrew/bin/glibtoolize /opt/homebrew/bin/libtoolize

> Found GNU Libtool version 2.4.7

[…]

> checking for __gmp_get_memory_functions in -lgmp... no
> configure: error: GMP version >= 4.2.0 needed, see http://gmplib.org

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -47,6 +47,10 @@ jobs:
     steps:
       - name: Check out
         uses: actions/checkout@v4
+      - name: Set Xcode
+        uses: maxim-lobanov/setup-xcode@v1
+        with:
+          xcode-version: latest-stable
       - name: Install prerequisites
         run: |
           brew update
@@ -56,8 +60,15 @@ jobs:
           brew upgrade autoconf
       - name: Configure
         run: |
+          if [[ $(uname -m) == 'arm64' ]]; then
+             GMP=/opt/homebrew/opt/gmp
+             MPFR=/opt/homebrew/opt/mpfr
+          else
+             GMP=/usr/local/include/gmp
+             MPFR=/usr/local/include/mpfr
+          fi
           ./autogen.sh
-          ./configure --disable-static $CONFIGURE_OPTS
+          ./configure --disable-static --with-gmp=$GMP --with-mpfr=$MPFR $CONFIGURE_OPTS
       - name: Compile
         run: make -j $JOBS
 

